bindings/node/lib/autoEncrypter.js

```diff
@@ -91,7 +92,7 @@ module.exports = function (modules) {
      */
     constructor(client, options) {
       this._client = client;
-      this._bson = options.bson || client.topology.bson;
+      this._bson = BSON || options.bson || client.topology.bson;
```

Comment:
when would `modules.mongodb.BSON` actually be undefined? I'm a bit confused about the order of the fallbacks here, especially since in the test files below we are defining the import unconditionally as mongodb.BSON - why is it ok to always expect it to be defined in tests but not in this code?

also, can we use the one subtask per PR convention we've established to make review and merge tracking easier? it would also be good to make sure we have a subtask for bumping that dependency that's called out in the description

Reply:
The tests are strange.  The behavior of the tests will (generally) depend on the version of the driver that the bindings are tested with because the fle classes are initialized with an instance of the mongodb module.  I think that the current setup is fine, because it does actually (for the most part, with some exceptions) test the behavior that we want.

We do generally only test with the version in dev dependencies in mongodb-client-encryption (now 4.8.1) and latest in the driver though.

There are other issues with the tests as well I believe, see my other comment

Reply:
`modules.mongodb.BSON` will always be undefined when using a driver version < 4.9.0 so we need to fallback first to the case where we passed it in the options then to the topology on the client which is the last resort. So 4.9.0 will solve the connected client requirement in the parent ticket but any version before 4.9.0 will still require it.

This ticket is now sub-task NODE-4556 and I've updated the description.